pkg/syncer/syncer.go

```diff
@@ -199,7 +199,11 @@ func (r *SyncController) Reconcile(ctx context.Context, vReq reconcile.Request)
 	if vObj != nil && pObj != nil {
 		// make sure the object uid matches
 		pAnnotations := pObj.GetAnnotations()
-		if !r.options.DisableUIDDeletion && pAnnotations[translate.UIDAnnotation] != "" && pAnnotations[translate.UIDAnnotation] != string(vObj.GetUID()) {
+
+		if !r.config.Sync.ToHost.Namespaces.Enabled && // if we're running with namespace sync enabled, we don't want to remove any host object based on UID because they might have been previously imported by other vclusters.
```

Comment:
What if a pod gets super quickly created and then deleted again? Wouldn't this cause issues when namespace sync is enabled as the pod would get outdated on the host? Basically this:
1. Create pod in vCluster
2. Sync pod to Host Cluster
3. Super quickly delete & create pod in vCluster
4. Now host cluster has a pod with that name and vCluster, but the spec could be different.
How do we solve this?

Reply:
then we probably should just explicitly cleanup objects when removing vCluster instead of overwriting. this way we don't need to change UID checks logic.